services/api/routes/authenticated.js

```diff
@@ -157,6 +158,112 @@ var routes = [
+                  ),
+                  title: Joi.string().max(256).required()
+                })
+              }),
+              Joi.object().keys({
+                type: Joi.string().valid('pages').required(),
+                method: Joi.string().valid('create').required(),
+                data: Joi.object().keys({
+                  projectId: Joi.alternatives().try(
+                    Joi.number().integer().required(),
+                    Joi.string().regex(/^\$(\d+)\.(.*)$/)
+                  ),
+                  x: Joi.number().integer().required(),
+                  y: Joi.number().integer().required(),
+                  styles: Joi.object().default({}).optional()
+                })
+              }),
+              Joi.object().keys({
+                type: Joi.string().valid('pages').required(),
+                method: Joi.string().valid('update').required(),
+                data: Joi.object().keys({
+                  id: Joi.alternatives().try(
+                    Joi.number().integer().required(),
+                    Joi.string().regex(/^\$(\d+)\.(.*)$/)
+                  ),
+                  x: Joi.number().integer().required(),
+                  y: Joi.number().integer().required(),
+                  styles: Joi.object().default({})
+                })
+              }),
+              Joi.object().keys({
+                type: Joi.string().valid('elements').required(),
+                method: Joi.string().valid('create').required(),
+                data: Joi.object().keys({
+                  pageId: Joi.alternatives().try(
+                    Joi.number().integer().required(),
+                    Joi.string().regex(/^\$(\d+)\.(.*)$/)
+                  ),
+                  type: Joi.string().required(),
+                  styles: Joi.object().default({}),
+                  attributes: Joi.object().default({})
+                })
+              }),
+              Joi.object().keys({
+                type: Joi.string().valid('elements').required(),
+                method: Joi.string().valid('update').required(),
+                data: Joi.object().keys({
+                  id: Joi.alternatives().try(
+                    Joi.number().integer().required(),
+                    Joi.string().regex(/^\$(\d+)\.(.*)$/)
+                  ),
+                  styles: Joi.object().default({}),
+                  attributes: Joi.object().default({})
+                })
+              }),
+              Joi.object().keys({
+                type: Joi.string().valid('projects', 'pages', 'elements').required(),
+                method: Joi.string().valid('remove').required(),
+                data: Joi.object().keys({
+                  id: Joi.alternatives().try(
+                    Joi.number().integer().required(),
+                    Joi.string().regex(/^\$(\d+)\.(.*)$/)
+                  )
+                })
+              })
+            ])
+          ).min(1).max(1000)
```

Comment:
neither do I. I had meant to have conversations about this.

On one hand, no limit sounds like we're begging to be DOS'd, but if it's too low, we'll end up having to increase it or workaround things in the app.